HerPublicWebsite.BusinessLogic/Services/CsvFileCreator/CsvFileCreator.cs

```diff
@@ -10,26 +10,149 @@ namespace HerPublicWebsite.BusinessLogic.Services.CsvFileCreator;
-        };
+    public MemoryStream CreateReferralRequestOverviewFileData(IEnumerable<ReferralRequest> referralRequests)
+    {
+        var rows = referralRequests.Select(rr => new CsvRowReferralCodes(rr));
+        return GenerateCsvMemoryStreamFromFileRows(rows);
+    }
 
-        using var writeableMemoryStream = new MemoryStream();
-        using var streamWriter = new StreamWriter(writeableMemoryStream, Encoding.UTF8);
-        using var csvWriter = new CsvWriter(streamWriter, csvConfiguration);
-        {
-            csvWriter.WriteRecords(rows);
-            csvWriter.Flush();
-            return new MemoryStream(writeableMemoryStream.GetBuffer(), 0, (int)writeableMemoryStream.Length, false);
+    public MemoryStream CreateReferralRequestFollowUpFileData(IEnumerable<ReferralRequest> referralRequests)
+    {
+        var rows = referralRequests
+            .GroupBy(rr => rr.CustodianCode)
+            .GroupBy(groupingByLa =>
+                LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[groupingByLa.Key].Consortium)
+            .SelectMany(groupingByConsortium =>
+                {
+                    var consortiumReferrals = groupingByConsortium.SelectMany(g => g).ToList();
+                    var consortiumStatistics = new ConsortiumStatistics(consortiumReferrals);
+                    return groupingByConsortium.Select(groupingByLa => new CsvRowLaDownloadInformation(groupingByLa, consortiumStatistics));
+                }
+            );
+        return GenerateCsvMemoryStreamFromFileRows(rows);
+    }
+
+    private class CsvRowReferralCodes
+    {
+        [Index(0)]
+        [Name("Consortium")]
+        public string Consortium { get; set; }
+        
+        [Index(1)]
+        [Name("Local Authority")]
+        public string LocalAuthority { get; set; }
+        
+        [Index(2)]
+        [Name("Referral Code")]
+        public string ReferralCode { get; set; }
+        public CsvRowReferralCodes(ReferralRequest request){
+            Consortium =  LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[request.CustodianCode].Consortium;
+            LocalAuthority =  LocalAuthorityData.LocalAuthorityDetailsByCustodianCode[request.CustodianCode].Name;
+            ReferralCode = request.ReferralCode;
+        }
+    }
+    
+    private class ConsortiumStatistics
+    {
+        public bool AllConsortiumReferralsDownloaded { get; set; }
+        public int NumberUndownloadedConsortiumReferrals { get; set; }
+        public float PercentageUndownloadedConsortiumReferrals { get; set; }
+        public bool AllConsortiumReferralsContacted { get; set; }
+        public int NumberUncontactedConsortiumReferrals { get; set; }
+        public float PercentageUncontactedConsortiumReferrals { get; set; }
+        public ConsortiumStatistics(List<ReferralRequest> referralRequests){
+            NumberUndownloadedConsortiumReferrals = referralRequests.Sum(rr => rr.ReferralWrittenToCsv ? 0 : 1);
```

Comment:
I think this would be clearer as `referralRequests.Count(rr => !rr.ReferralWrittenToCsv)`? (And similarly with all the other sums below)